apk: libudev dependency for usbutils not automatically installed

### Describe the bug

When using the imagebuilder to create the image for my device, it fails on installing the `usbutils` package due to this error:

```
Installing packages...
ERROR: unable to select packages:
  libudev (virtual):
    note: please select one of the 'provided by'
          packages explicitly
    provided by: libudev-zero
    required by: usbutils-017-r1[libudev]
make[2]: *** [Makefile:229: package_install] Error 2
make[1]: *** [Makefile:160: _call_image] Error 2
make: *** [Makefile:332: image] Error 2
```

When I add the `libudev-zero` dependency manually to the `PACKAGES` list for the `make` command, the imagebuilder finishes.

It seems counterintuitive, that `libudev-zero` is not automatically selected and installed as dependency, when it is the only provider available. As it has been with opkg before.

After discussing the issue on IRC it seems that this behavior is not intended, hence the bug report.

### OpenWrt version

r28091-77cfe8fd15

### OpenWrt release

SNAPSHOT

### OpenWrt target/subtarget

mediatek/filogic

### Device

Bananapi BPI-R4

### Image kind

Official downloaded image

### Steps to reproduce

Download and extract [imagebuilder](https://downloads.openwrt.org/snapshots/targets/mediatek/filogic/openwrt-imagebuilder-mediatek-filogic.Linux-x86_64.tar.zst)

Run build with `make image PROFILE=bananapi_bpi-r4 PACKAGES="usbutils"`

### Actual behaviour

Build fails due to error.

### Expected behaviour

Build completes successfully.

### Additional info

_No response_

### Diffconfig

_No response_

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

--- a/include/package-pack.mk
+++ b/include/package-pack.mk
@@ -359,7 +359,7 @@ else
 			) \
 		), \
 		$$(prov) )" \
-	  $(if $(DEFAULT_VARIANT),--info "provider-priority:100") \
+	  $(if $(DEFAULT_VARIANT),--info "provider-priority:100",$(if $(PROVIDES),--info "provider-priority:1")) \
 	  --script "post-install:$$(ADIR_$(1))/post-install" \
 	  --script "pre-deinstall:$$(ADIR_$(1))/pre-deinstall" \
 	  --info "depends:$$(foreach depends,$$(subst $$(comma),$$(space),$$(subst $$(space),,$$(subst $$(paren_right),,$$(subst $$(paren_left),,$$(Package/$(1)/DEPENDS))))),$$(depends))" \

